### PR TITLE
implement per variant default choice card contribution frequency

### PR DESCRIPTION
**Pull request by michaelbjacobson**
## What does this change?
Allows per variant configuration of default contribution amount choice card.

### CODE REVIEW

File: app/models/EpicTests.scala [hunk 1]
```diff
@@ -23,6 +23,15 @@ object TickerCountType extends Enum[TickerCountType] with CirceEnum[TickerCountT
   case object people extends TickerCountType
 }
 
+sealed trait ContributionFrequency extends EnumEntry
```
Comment (tomrf1) [inline app/models/EpicTests.scala:26]: btw circe can now do enums with slightly neater syntax, see https://github.com/guardian/support-admin-console/blob/main/app/models/SupportFrontendSwitches.scala#L8
I realise lots of places still do it this way though!
E.g.
```
sealed trait ContributionFrequency
case object ONE_OFF extends ContributionFrequency
case object MONTHLY extends ContributionFrequency
case object ANNUAL extends ContributionFrequency
```
and then further down next to the Encoders/Decoders:
```
implicit val stateDecoder: Decoder[ContributionFrequency] = deriveEnumerationDecoder[ContributionFrequency]
implicit val stateEncoder: Encoder[ContributionFrequency] = deriveEnumerationEncoder[ContributionFrequency]
```

**Review by tomrf1 — APPROVED** [review]
looks good. Will this work with the new way of rendering choice cards from an iframe?

**Comment by michaelbjacobson** [comment]
> looks good. Will this work with the new way of rendering choice cards from an iframe?

I've no idea, I think I must have been off when that change was made! I manually tested this change in `CODE` using the web preview functionality and it all worked perfectly. Do you know who did the iframe work?

**Comment by tomrf1** [comment]
> > looks good. Will this work with the new way of rendering choice cards from an iframe?
> 
> I've no idea, I think I must have been off when that change was made! I manually tested this change in `CODE` using the web preview functionality and it all worked perfectly. Do you know who did the iframe work?

I don't think the iframe stuff is live yet, @TomPretty will know more!

File: app/models/EpicTests.scala (hunk 1 above)
Comment (michaelbjacobson) [reply app/models/EpicTests.scala:26]: @tomrf1 thanks, this is v good to know for the future, but I realised that this type had already been created elsewhere in the codebase and named `ContributionType`.

Comment (tomrf1) [reply app/models/EpicTests.scala:26]: oh, good spot!

**Pull request merged**